### PR TITLE
Subject Controllers and Views

### PR DESCRIPTION
**Pull request by leduytungframgia**
Subject Controllers and Views

### CODE REVIEW

File: app/controllers/subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def create
+    @subject = Subject.new subject_params
+    if @subject.save
+      flash[:success] = "Subject created!"
+      redirect_to @subject
+    end
+  end
+
+  def new
+  	@subject = Subject.new
```
Comment (PhamVanTung) [inline app/controllers/subjects_controller.rb]: indent

File: app/views/subjects/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<% provide :title, @subject.name %>
+  
+    <div class="panel panel-info">
```
Comment (PhamVanTung) [inline app/views/subjects/show.html.erb]: indent

File: app/controllers/subjects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  def destroy
+  end
+
+  def show
+    @subject = Subject.find params[:id]
+  end
+
+  private
+  def subject_params
+    params.require(:subject).permit(:name, :instruction)
```
Comment (vjnhat) [inline app/controllers/subjects_controller.rb]: remove () => params.require(:subject).permit :name, :instruction

File: app/views/subjects/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<% provide :title, @subject.name %>
+<div class="panel panel-info">
+    <div class="alert alert-info" role="alert"><span class="user">
+      <h3><%= link_to @subject.name, @subject %></h3>
+      <span class="timestamp">
+      Posted <%= time_ago_in_words @subject.created_at %> ago.
+
+      <%= link_to "Delete", @entry, method: :delete,
+                                   data: { confirm: "You sure?" } %>
```
Comment (PhamVanTung) [inline app/views/subjects/show.html.erb]: indent

File: app/controllers/subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+class SubjectsController < ApplicationController
+  def create
+    @subject = Subject.new subject_params
+      if @subject.save
```
Comment (chuanhtuan) [inline app/controllers/subjects_controller.rb]: indent

File: app/controllers/subjects_controller.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
+class SubjectsController < ApplicationController
+  def new
+    @subject = Subject.new
+  end
+
+  def create
+    @subject = Subject.new subject_params
+    if @subject.save
+      flash[:success] = "Subject created!"
```
Comment (dieunb) [inline app/controllers/subjects_controller.rb]: i18n

File: app/views/subjects/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<% provide :title, 'New Subject' %>
+<%= form_for @subject do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <div class="field">
+    <%= f.text_field :name, placeholder: "Compose new name..." %>
```
Comment (dieunb) [inline app/views/subjects/new.html.erb]: i18n

File: app/views/subjects/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  </div>
+  <div class="field">
+    <%= f.text_area :instruction, placeholder: "Compose new instruction..." %>
```
Comment (dieunb) [inline app/views/subjects/new.html.erb]: i18n

File: app/views/subjects/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  </div>
+  <div class="field">
+    <%= f.text_field :start_date, placeholder: "Compose start date..." %>
```
Comment (dieunb) [inline app/views/subjects/new.html.erb]: i18n

File: app/views/subjects/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  </div>
+  <div class="field">
+    <%= f.text_field :end_date, placeholder: "Compose end date..." %>
```
Comment (dieunb) [inline app/views/subjects/new.html.erb]: i18n

File: app/views/subjects/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<% provide :title, @subject.name %>
+<div class="panel panel-info">
+    <div class="alert alert-info" role="alert"><span class="user">
+      <h3><%= link_to @subject.name, @subject %></h3>
+      <span class="timestamp">
+      Posted <%= time_ago_in_words @subject.created_at %> ago.
+
+      <%= link_to "Delete", @entry, method: :delete,
+                                   data:{confirm: "You sure?"} %>
```
Comment (vjnhat) [inline app/views/subjects/show.html.erb]: missing space after data:

**Pull request closed without merging**